README.md

```diff
@@ -13,13 +13,74 @@ issue requests and receive responses in Standard Interchange Protocol v2 (SIP2).
+
+```bash
+$ java -jar edge-sip2-fat.jar -conf '{"port":1234,"okapiUrl":"https://folio-snapshot-okapi.aws.indexdata.com","tenant":"diku"}'
+```
+
+|Config option|Type|Description|
+|-------------|----|-----------|
+|`port`|int|The port the module will use to bind, typically 1024 < port < 65,535.|
+|`okapiUrl`|string|The URL of the Okapi server used by FOLIO.|
+|`tenant`|string|The FOLIO assigned tenant ID. Multi-tenant support TBD.|
+|`fieldDelimiter`|string|The character that the self service kiosk will use when encoding SIP messages. Defaults to "\|".|
+|`errorDetectionEnabled`|boolean|Indicates whether or not the self service kiosk will be using SIP error detection in messages sent to and from this module. Defaults to "false".|
+|`charset`|string|The character set SIP messages must be encoded with when sent and received by the self service kiosk. The charset must be defined as a "Canonical Name for java.nio API". See: [Supported Encodings](https://docs.oracle.com/javase/8/docs/technotes/guides/intl/encoding.doc.html). Default is "IBM850".|
+|`messageDelimiter`|string|The character sequence that indicates the end of a single SIP message. This is available in case the self check kiosk is not compliant with the SIP specification. The default is "\\r"|
+|`netServerOptions`|JSON object|Configuration options for the server. These are Vertx options and are numerous. See: [NetServerOptions](https://vertx.io/docs/apidocs/io/vertx/core/net/NetServerOptions.html).|
+
 ## Security
 
-TBD (most likely stunnel)
+The SIP protocol does not consider security apart from providing the possibility of SC/ACS negotiated encryption algorithm for the password and user ID, which seems unlikely to be used. All communication over the wire, via TCP, is plain text. This may be fine in an environment that is locked down in some way. However, FOLIO, along with this module, can be hosted in the cloud and plain text communication is not acceptable.
 
-## Configuration
+We **_strongly_** recommend securing communication from the SC to edge-sip2. To this end, edge-sip2 can be configured to use TLS to encrypt communication. This requires the SC to communicate with TLS as well. It is our understanding that most self service kiosks do not have this ability natively and third party solutions must be employed. One such solution is `stunnel`.
+
+A typical `stunnel` deployment will involve installing the `stunnel` service either on the SC or a machine that is locked down with the SC and provides a port that the SC will be configured to connect to for SIP communication. The SIP commands are then sent to and received from the `stunnel` port unencrypted. Communication from the `stunnel` to the TLS termination end point will be encrypted. There are several ways to terminate TLS and the advantages/disadvantages of these is out of scope for this document. Here, we will focus on enabling TLS termination via the edge-sip2 module. This is done via simple launch configuration options.
+
```

Comment:
> The SIP commands are then sent to and received from the `stunnel` port unencrypted.

To and from the SC application?

> TLS termination end point

Termination endpoint == ACS?

Reply:
> To and from the SC application?

Most likely yes. They may have a proxy in place or something else between the SC and the `stunnel` application.

> Termination endpoint == ACS?

Could be the ACS/edge-sip2. It could be a load balancer or some other SSL offloading service. It could even be `stunnel` in server mode in front of edge-sip2.

I was trying to be a bit vague here since deployment environments, especially related to SSL/TLS will likely vary quite a bit.

Reply:
ok